onnxruntime/core/providers/cuda/tensor/concat.cc

```diff
@@ -59,24 +58,20 @@ Status Concat::ComputeInternal(OpKernelContext* ctx) const {
     concat_sizes_range[i] += concat_sizes_range[i - 1];
   }
 
-  CudaAsyncBuffer<int64_t> concat_sizes_gpu(this, concat_sizes);
-  CudaAsyncBuffer<int64_t> axis_dimension_input_output_mapping_gpu(this, axis_dimension_input_output_mapping);
-  CudaAsyncBuffer<int64_t> concat_sizes_range_gpu(this, concat_sizes_range);
-  concat_sizes_gpu.CopyToGpu();
-  axis_dimension_input_output_mapping_gpu.CopyToGpu();
-  concat_sizes_range_gpu.CopyToGpu();
-  input_ptr.CopyToGpu();
+  TArray<int64_t> concat_sizes_gpu(concat_sizes);
```

Comment:
>concat_sizes [](start = 35, length = 12)

concat takes variadic inputs, so the default TArray size may not be enough for some models.